app/controllers/api/v1/reservations_controller.rb

```diff
@@ -0,0 +1,54 @@
+module Api
+  module V1
+    class ReservationsController < ApplicationController
+      load_and_authorize_resource
+      before_action :set_reservation, only: %i[show update destroy]
+
+      def index
+        @reservations = current_user.reservations
+        render json: @reservations
+      end
+
+      def show
+        render json: @reservation
+      end
+
+      def create
+        @reservation = current_user.reservations.build(reservation_params)
+
+        if @reservation.save
```

Comment:
Since the `current_user` method has changed to `current_api_v1_user`, please update the `current_user` methods in the reservation controller to `current_api_v1_user` to fix the bugs that are caused by the wrong method name